### DISCUSSION TITLE
Disable text editing in preview mode

### DISCUSSION THREAD

**Discussion by glaprida** [GrapesJS/grapesjs#5509] [Q&A] [answered]
Right now, I if double click on a text element, I can edit its text even if I am in preview mode.

Is there a way to prevent this without going through each element and changing its `editable` value?
If I do that, then I would need to change it back when I exit preview mode...

Is there a way to do it globally? 
I tried locking the wrapper element, but that sets `pointer-events:none` and I want for things to be hoverable, and buttons to work.

In my opinion text elements should be editable on preview mode by default. Maybe something that can be 'fixed' in future releases?

**Comment by glaprida** [comment]
I also tried setting `editable: false` and `propagate: ['editable']` on the wrapper... but (as the docs say) it only affects NEW appended children.

Maybe there's a way to 'refresh' the wrapper and have it propagate through its old children?

**Comment by artf** [comment, marked as answer]
Something like this should work
```js
const fn = (ev) => ev.stopPropagation();

editor.on('run:core:preview', () => {
  const frameDoc = editor.Canvas.getDocument();
  frameDoc.addEventListener('dblclick', fn, { capture: true })
});

editor.on('stop:core:preview', () => {
  const frameDoc = editor.Canvas.getDocument();
  frameDoc.removeEventListener('dblclick', fn, { capture: true })
});
```
